File: Samples for Solid Edge/[C#]-Samples for Solid Edge/C#/SolidEdge.UnitsOfMeasure/MainForm.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Runtime.InteropServices;
using System.Text;
using System.Windows.Forms;

namespace SolidEdge.UnitsOfMeasure
{
    public partial class MainForm : Form
    {
        public MainForm()
        {
            InitializeComponent();
        }

        private void MainForm_Load(object sender, EventArgs e)
        {
            BeginInvoke(new MethodInvoker(MainForm_Load_Async));
        }

        /// <summary>
        /// Asynchronous version of MainForm_Load.
        /// </summary>
        private void MainForm_Load_Async()
        {
            SolidEdgeFramework.Application application = null;
            SolidEdgeFramework.Documents documents = null;

            // Connect to or start Solid Edge.
            application = SolidEdgeUtils.Connect(true);

            // Ensure Solid Edge is visible.
            application.Visible = true;

            // Get a reference to the Documents collection.
            documents = application.Documents;

            if (documents.Count == 0)
            {
                documents.Add("SolidEdge.PartDocument");
            }

            externalPropertyGrid.SelectedObject = new ExternalExample();
            internalPropertyGrid.SelectedObject = new InternalExample();
        }

        private void exitToolStripMenuItem_Click(object sender, EventArgs e)
        {
            Close();
        }

        private void externalPropertyGrid_PropertyValueChanged(object s, PropertyValueChangedEventArgs e)
        {
            externalPropertyGrid.Refresh();
        }

        private void internalPropertyGrid_PropertyValueChanged(object s, PropertyValueChangedEventArgs e)
        {
            internalPropertyGrid.Refresh();
        }
    }
}

File: Samples for Solid Edge/[VB]-Samples for Solid Edge/VB/SolidEdge.UnitsOfMeasure/MainForm.vb
Imports System.ComponentModel
Imports System.Runtime.InteropServices
Imports System.Text

Namespace SolidEdge.UnitsOfMeasure
	Partial Public Class MainForm
		Inherits Form

		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub MainForm_Load(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load
			BeginInvoke(New MethodInvoker(AddressOf MainForm_Load_Async))
		End Sub

		''' <summary>
		''' Asynchronous version of MainForm_Load.
		''' </summary>
		Private Sub MainForm_Load_Async()
			Dim application As SolidEdgeFramework.Application = Nothing
			Dim documents As SolidEdgeFramework.Documents = Nothing

			' Connect to or start Solid Edge.
			application = SolidEdgeUtils.Connect(True)

			' Ensure Solid Edge is visible.
			application.Visible = True

			' Get a reference to the Documents collection.
			documents = application.Documents

			If documents.Count = 0 Then
				documents.Add("SolidEdge.PartDocument")
			End If

			externalPropertyGrid.SelectedObject = New ExternalExample()
			internalPropertyGrid.SelectedObject = New InternalExample()
		End Sub

		Private Sub exitToolStripMenuItem_Click(ByVal sender As Object, ByVal e As EventArgs) Handles exitToolStripMenuItem.Click
			Close()
		End Sub

		Private Sub externalPropertyGrid_PropertyValueChanged(ByVal s As Object, ByVal e As PropertyValueChangedEventArgs) Handles externalPropertyGrid.PropertyValueChanged
			externalPropertyGrid.Refresh()
		End Sub

		Private Sub internalPropertyGrid_PropertyValueChanged(ByVal s As Object, ByVal e As PropertyValueChangedEventArgs) Handles internalPropertyGrid.PropertyValueChanged
			internalPropertyGrid.Refresh()
		End Sub
	End Class
End Namespace
